### DISCUSSION TITLE
`output_dir` is not created if not exists

### DISCUSSION THREAD

**Discussion by thii** [SwiftGen/SwiftGen#899] [Ideas]
`output_dir` specified in the config file is not automatically created if it doesn't exist before running `swiftgen`. I'm not sure if this is intentional for any reason but if not, it would be nice if `swiftgen` automatically creates it. Currently we're having to wrap `swiftgen` in a script to do this.

**Comment by djbe** [comment]
We purposefully do not touch the directory structure, particularly if a user made a mistake and are suddenly creating directories (or full blown deep structures) in some unknown location. Our `config lint` command checks if the paths are available, so you'll get an error there.

Do note that you can add a `.gitkeep` file to an empty folder and commit that, to preserve directory structures when working with git/hg/...

**Comment by thii** [comment]
What do you think about adding an optional flag to allow this behavior?

Our use-case might be a bit uncommon—the generate code doesn't live in the same directory structure with the version control source code, and tools are required to create files/directories on their own if they declare them as their outputs.

**Reply by djbe** [reply]
Might be a good idea, but I'm not sure where to put in the config file/structure.
- At the top level? (next to existing cmds and path options)
- Add a top-level `options`, where we also move the in/out paths?
